### ISSUE TITLE
Compile error: cannot find symbol

### ISSUE THREAD

**Issue by newbananaclub** [wf9a5m75/phonegap-googlemaps-plugin#512]
Hi everyone, 

I am building an Ionic app and tried to implement this plugin into my app. I have followed the tutorial on https://github.com/wf9a5m75/phonegap-googlemaps-plugin/wiki/Tutorial-for-Mac and it works great on iOS. However when I try to compile and run on Android, I get the following error: 
![screen shot 2015-04-26 at 10 20 24](https://cloud.githubusercontent.com/assets/9978160/7335615/359b996a-ebfe-11e4-9cca-6b24fdc4b47f.png)

I googled the problem and didn't have any progress, I tried the following steps and still no luck: 

```
cordova plugim rm plugin.google.maps
cordova plugin rm com.google.playservices
cordova plugin add https://github.com/wf9a5m75/google-play-services#v23
cordova plugin add plugin.google.maps --variable API_KEY_FOR_ANDROID="YOUR_ANDROID_API_KEY_IS_HERE" --variable API_KEY_FOR_IOS="YOUR_IOS_API_KEY_IS_HERE"
```

Here is my list of plugins:

```
com.google.playservices 23.0.0 "Google Play Services for Android"
com.googlemaps.ios 1.9.2 "Google Map iOS SDK for Cordova"
com.ionic.keyboard 1.0.4 "Keyboard"
org.apache.cordova.console 0.2.13 "Console"
org.apache.cordova.device 0.3.0 "Device"
plugin.google.maps 1.2.5 "phonegap-googlemaps-plugin"
plugin.http.request 1.0.4 "phonegap-http-request"
```

Any help is much appreciated!

**Comment by Hirbod** [comment]
Did you install cordova 4.0? I think it is not compatible yet

**Comment by colloquet** [comment]
Hi Hirbod, 

Thanks for the prompt reply!

I just checked and I am using Cordova 5.0, does that mean I will have to wait for the plugin to update?

Thanks!

**Comment by Hirbod** [comment]
Yes, as far as I know, the plugin does not work on the Cordova android 4.0 platform

**Comment by colloquet** [comment]
Right, Thanks very much!

**Comment by jskrzypek** [comment]
I had this error but it is fixed on the test branch. To use it you need to run these commands: 
```sh
cordova plugin rm plugin.google.maps
cordova plugin add https://github.com/wf9a5m75/phonegap-googlemaps-plugin@test --variable API_KEY_FOR_ANDROID="<put your API key in here>"
```

**Comment by jskrzypek** [comment]
Never mind about that, it built successfully only because it wasn't actually grabbing the plugin since I was using `@test` instead of `#test`. I'm getting the same error.

**Comment by newbananaclub** [comment]
Guess we will have to wait until it updates :(

**Comment by jskrzypek** [comment]
Yup. I tried digging into the native code, definitely not my bailiwick, but that was a headache and a half and went nowhere. My suggesstion is if you need this plugin stick to cordova 4.x for now

**Comment by leomylonas** [comment]
I'm also running Cordova 5.0.0 with android project 4 and require an update ASAP.

**Comment by Hirbod** [comment]
There a many changes currently, and @wf9a5m75 is very busy at the moment. We need to be patient right now till he find the time to upgrade the plugin for compatibillity

**Comment by leandroz** [comment]
Hey guys lets help @wf9a5m75 with the update, is an Open Source project, he doesn't need to have all the responsibility to update it.

The ScrollEvent is deprecated, see here:
https://github.com/apache/cordova-android/commit/581252febc2ca8c2ecbe7077047a34aae0ab25b5

webView.getParent() maybe this help, is from AppGyver about this plugin too.
https://github.com/AppGyver/steroids/issues/588
CordovaWebView in Android v4 is an implementation file, and the implementation is in CordovaWebViewImp and it doesn't extend WebView anymore so the getParent method is not available.
But they are using it like: ````engine.getView().getParent();````.
engine is https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/engine/SystemWebViewEngine.java that is working as a glue between the real WebView, the ````getView()```` method returns the webView instance.

I think that probably the other methods is the same because they WebView methods are not available anymore directly in CordovaWebView.
Hope this help.

**Comment by jskrzypek** [comment]
Hey @leandroz I'm totally on board with working to make the changes you suggest. Have you opened a fork for the issue? I'm happy to help work on it though I don't have *too* much spare time.

I think I have working code to do the job the ScrollEvent was doing, though maybe if we don't actually have a webView in CordovaWebView, ViewTreeObserver might not work correctly...

Basically it's like this:

```java
final ViewTreeObserver.OnScrollChangedListener scrollListener = new ViewTreeObserver.OnScrollChangedListener() {
      // @Override
      public void onScrollChanged(int l, int t, int oldl, int oldt) {
        
        if (mPluginLayout != null) {
          mPluginLayout.scrollTo(oldl, oldt);
          if (mapDivLayoutJSON != null) {
            try {
              float divW = contentToView(mapDivLayoutJSON.getLong("width"));
              float divH = contentToView(mapDivLayoutJSON.getLong("height"));
              float divLeft = contentToView(mapDivLayoutJSON.getLong("left"));
              float divTop = contentToView(mapDivLayoutJSON.getLong("top"));
      
              mPluginLayout.setDrawingRect(
                  divLeft,
                  divTop - oldt, 
                  divLeft + divW, 
                  divTop + divH - oldt);
            } catch (JSONException e) {
              e.printStackTrace();
            }
          }
        }
      }
    };

    webView.getView().getViewTreeObserver().addOnScrollChangedListener(scrollListener);
```

**Comment by Davide95** [comment]
Hello to everybody, I've the same issue.
I have few questions:
How can I downgrade to Apache Cordova 4.x?
I'm using Ionic framework, are they incompatibilities to do the downgrade?
Are there future plans to upgrade this plugin?

Thanks in advance, sorry to bother you.

**Comment by jskrzypek** [comment]
Hi @Davide95, downgrading is very simple using npm. Simply run the command in your shell: 

```sh
npm install -g cordova@4.3.0
```
Then if you have projects that have been upgraded to use `cordova-android@4.0.0`—the new, as-yet-incompatible android version—navigate to their root project directories where the `www` is located and run
```sh
cordova platform update android@3.7.1
```

**Comment by teuteuguy** [comment]
Hi everyone, 
has there been any progress on updating this issue for cordova 5.0.0 ?
Thanks

**Comment by Davide95** [comment]
Hi, there is [this](https://github.com/Nipher/phonegap-googlemaps-plugin) but I've never tried it
Let me know if it works :)

**Comment by teuteuguy** [comment]
Hi Davide95,

Yup, it works. I manage to build the app, and run it on Android.

I cannot say if there are any side-effects because this is the first time we are running this on Android, and also we are still in the testing phase of using this plugin.

For now, I'll simply state that it works.

What would be the next steps for getting this merged ?

Thanks

**Comment by dortzur** [comment]
Downgraded :( 
+1 for this issue

**Comment by Davide95** [comment]
The problem appears only when it is built for android? Or iOs has the same problem?

**Comment by dortzur** [comment]
Android when bumped to version 4.0
iOS doesn't have this but they didn't bump to version 4.0 yet.

**Comment by Davide95** [comment]
Sorry but I've not understood. I ve cordova 5, so I can't use it for iOs?

**Comment by dortzur** [comment]
Cordova team split their CLI version (5.0) and their platform versions.
Check it out here

Android (4.0)
https://github.com/apache/cordova-android/releases

iOS (3.8.0)
https://github.com/apache/cordova-ios/releases

CLI (5.0)
https://github.com/apache/cordova-cli/releases

When iOS version 4.0 is released, it might have breaking changes, but that
hasn't happened yet, so only Android has breaking changes as of now.






On Mon, May 25, 2015 at 8:27 PM, Davide Riva <notifications@github.com>
wrote:

> Sorry but I've not understood. I ve cordova 5, so I can't use it for iOs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wf9a5m75/phonegap-googlemaps-plugin/issues/512#issuecomment-105273963>
> .
>



-- 
Dor Tzur
Phone: +972.54.7421313
LinkedIn: http://www.linkedin.com/in/dortzur
Facebook: *https://www.facebook.com/dortzur1
<https://www.facebook.com/dortzur1>*

**Comment by Davide95** [comment]
Thank you so much!

**Comment by dortzur** [comment]
Thank you for this awesome plugin ;)

On Mon, May 25, 2015 at 11:17 PM, Davide Riva <notifications@github.com>
wrote:

> Thank you so much!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wf9a5m75/phonegap-googlemaps-plugin/issues/512#issuecomment-105305519>
> .
>



-- 
Dor Tzur
Phone: +972.54.7421313
LinkedIn: http://www.linkedin.com/in/dortzur
Facebook: *https://www.facebook.com/dortzur1
<https://www.facebook.com/dortzur1>*

**Comment by Davide95** [comment]
I am not the author, he is @wf9a5m75 :)

**Comment by jskrzypek** [comment]
Hey I'm late to this a bit, but looking at the Nipher/phonegap-googlemaps-plugin branch they don't re-implement the scrollevent system that was removed in cordova 5/cordova-android 4 so the onScrollEvent won't work I think.

**Comment by jacksonkr** [comment]
I've tried both maps plugins and I'm still stuck with this atrocious error. Did anyone figure out a solution at all? I'm supposed to have a new APK by tomorrow.

Cordova CLI: 5.1.1
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.5.2
Ionic App Lib Version: 0.2.0
ios-deploy version: Not installed
ios-sim version: 3.1.1 
OS: Mac OS X Yosemite
Node Version: v0.12.1
Xcode version: Xcode 6.3.1 Build version 6D1002

**Comment by codebling** [comment]
@jacksonkr as mentioned above, if you need this for tomorrow the only solution is to downgrade Cordova to 4.x.x

**Comment by jacksonkr** [comment]
@codebling believe me I've been up and down this thread several times. Your suggestion definitely works for getting maps working again but it disables the whitelist plugin. I know that yesterday before this started happening that I was able to build android with both google maps and whitelist. I'm just going to have to uninstall whitelist (needed) until maps is fixed.

**Comment by codebling** [comment]
@jacksonkr this is going to sound harsh but the Github issue is not the right place to ask for support. The fact that the whitelist plugin does or doesn't work isn't germane to the bug and adding several pages worth of command line output just slows down everyone trying to follow the conversation. You can email me at my user name plus an x @gmail.com and I'll help you.

**Comment by 2WheelCoder** [comment]
Even Cordova 4.3.1 build is failing for me. If I remove the plugin it works fine. I tried 3.x.x too.

I've gotten through several errors to get here, such as installing ant, setting PATH variables and installing correct SDK version, but seem to be stuck on what to do from here. Any suggestions? Thanks!

Here's the error in 4.3.1:

```
BUILD FAILED
/Users/nickstevens/Library/Android/sdk/tools/ant/build.xml:716: The following error occurred while executing this line:
/Users/nickstevens/Library/Android/sdk/tools/ant/build.xml:730: Compile failed; see the compiler error output for details.

Total time: 3 seconds

/Users/nickstevens/Sites/postly_mobile/platforms/android/cordova/node_modules/q/q.js:126
                    throw e;
                          ^
Error code 1 for command: ant with args: debug,-f,/Users/nickstevens/Sites/postly_mobile/platforms/android/build.xml,-Dout.dir=ant-build,-Dgen.absolute.dir=ant-gen
ERROR building one of the platforms: Error: /Users/nickstevens/Sites/postly_mobile/platforms/android/cordova/build: Command failed with exit code 8
You may not have the required environment or OS to build this project
Error: /Users/nickstevens/Sites/postly_mobile/platforms/android/cordova/build: Command failed with exit code 8
    at ChildProcess.whenDone (/Users/nickstevens/.nvm/v0.10.38/lib/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:131:23)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:766:16)
    at Process.ChildProcess._handle.onexit (child_process.js:833:5)
```

**Comment by Juelcf** [comment]
Hi all,

I found a solution that worked for me to build with cordova 5.1.1

https://github.com/wf9a5m75/phonegap-googlemaps-plugin/issues/576

**Comment by svr93** [comment]
Hey!
http://www.java2s.com/Open-Source/Android_Free_Code/PhoneGap/Cordova/org_apache_cordovaScrollEvent_java.htm
What about using it as a hotfix (instead of org.apache.cordova.ScrollEvent) ?

**Comment by svr93** [comment]
![123456](https://cloud.githubusercontent.com/assets/6628541/9028097/81b59de6-397e-11e5-8db6-416db2de42b8.jpg)
cordova 5.0.0
plugin.google.maps 1.2.5

**Comment by Hirbod** [comment]
fixed in master (Version 1.2.7 and newer). Please reinstall.
